kube-client/src/config/file_config.rs

```diff
@@ -350,7 +350,17 @@ impl Kubeconfig {
     ///
     /// Panics if `KUBECONFIG` value contains the NUL character.
     pub fn from_env() -> Result<Option<Self>, KubeconfigError> {
-        match std::env::var_os(KUBECONFIG) {
+        Self::from_custom_env(KUBECONFIG)
+    }
+
+    /// Create `Kubeconfig` from a given environment variable.
+    /// Supports list of files to be merged.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the environment variable value contains the NUL character.
+    pub fn from_custom_env(env_name: &str) -> Result<Option<Self>, KubeconfigError> {
+        match std::env::var_os(env_name) {
```

Comment:
As much as it feels like a rare use-case to have people use different evars, **this way** of exporting two sides of the functionality in `from_env` that you have here, is probably the best split.

Tried to think of a few other splits where `::from_env` would parse the KUBECONFIG first, and maybe splits the paths, then maybe collects (and maybe even reads the files), before then calling a merger fn; but none of those options feel like they are actually giving us a useful split IO/non-IO split without making a very ugly interface.

That said, I'm kind of more positive on exporting the `merge` fn here than to have `kube` read a different non-standard evar as that feels very unexpected for kube to do (and something i expect to be an extremely rare use-case - unless I am missing something).

Reply:
100% understandable, it's a rare use-case indeed. Exporting merge would also work for me, so I created this other PR #1100

Reply:
Appreciate it. Have merged this and closed this PR. It's nicer to have fns that export documented parts of kubernetes algorithms be public.